libcron/include/libcron/Cron.h

```diff
@@ -149,27 +212,64 @@ namespace libcron
             if (t.calculate_next(clock.now()))
             {
                 tasks.push(t);
+                tasks.sort();
             }
             tasks.release_queue();
         }
 
         return res;
     }
 
-    template<typename ClockType, typename LockType>
-    void Cron<ClockType, LockType>::clear_schedules()
+    template<typename ClockType, typename LockType, typename NameScheduleMapType>
+    bool Cron<ClockType, LockType, NameScheduleMapType>::add_schedule(const NameScheduleMapType& name_schedule_map, Task::TaskFunction work)
+    {
+        bool res = false;
+        std::vector<Task> tasks_to_add;
+        tasks_to_add.reserve(name_schedule_map.size());
+
+        for (const auto& [name, schedule] : name_schedule_map)
+        {
+            auto cron = CronData::create(schedule);
+            res = cron.is_valid();
+            if (res)
+            {
+                Task t{std::move(name), CronSchedule{cron}, work };
+                if (t.calculate_next(clock.now()))
+                {
+                    tasks_to_add.push_back(std::move(t));
+                }
+            }
+            else 
+            {
+                break;
```

Comment:
I see I haven't added prohibition of break/continue to the guideline I linked earlier so no blame on you, but please don't use them, I consider them to be equivalent to goto-statements which are also prohibited. Yes, it is a contentious opinion, but it has worked well for me for decades so I'm not changing this rule :)